Tchap/Modules/Room/RoomViewController.m

```diff
@@ -1166,6 +1182,72 @@ - (void)restoreDiscussionIfNeed:(void (^)(BOOL success))onComplete
     }];
 }
 
+/**
+ Create a direct chat with given user.
+ */
+- (void)createDiscussionWithUser:(User*)user completion:(void (^)(BOOL success))onComplete
+{
+    MXWeakify(self);
+    
+    [self startActivityIndicator];
+    
+    [self.mainSession createRoom:nil
+                      visibility:kMXRoomDirectoryVisibilityPrivate
+                       roomAlias:nil
+                           topic:nil
+                          invite:@[user.userId]
+                      invite3PID:nil
+                        isDirect:YES
+                          preset:kMXRoomPresetTrustedPrivateChat
+                         success:^(MXRoom *room) {
+                             
+                             MXStrongifyAndReturnIfNil(self);
+                             
+                             MXKRoomDataSourceManager *roomDataSourceManager = [MXKRoomDataSourceManager sharedManagerForMatrixSession:self.mainSession];
+                             
+                             [roomDataSourceManager roomDataSourceForRoom:room.roomId create:YES onComplete:^(MXKRoomDataSource *roomDataSource) {
+                                 
+                                 [self stopActivityIndicator];
+                                 
+                                 if (roomDataSource.room.summary.isDirect == NO)
+                                 {
+                                     // TODO: Display an error, and retry to set room as direct otherwise quit the room
+                                     
+                                     NSLog(@"[RoomViewController] Fail to create room as direct chat");
+                                 }
+//                                 else
+//                                 {
+                                 // Set user to nil and now use RoomDataSource for updating view
+                                     self.discussionTargetUser = nil;
+                                     [self displayRoom:roomDataSource];
```

Comment:
Remark: the discussionTargetUser should be reseted in [displayRoom] method by default